### ISSUE TITLE
401 Unauthorized Error

### ISSUE THREAD

**Issue by samirkhan70** [natewallace/ApexNTLM#5]
Hi Nate,

I am executing below code.

String body = system.label.MAFLF_SPBody;
String ep = 'https://pprojectstest.xxx.ae/SF/_vti_bin/copy.asmx';
system.debug('system.label.MAFLF_SPBody-->'+system.label.MAFLF_SPBody);
HttpClient httpObj = new HttpClient('domain\\USERNXXXXXXX','PASSXXXX',true); 
HttpRequest req = new HttpRequest();
req.setBody(body);
req.setMethod('POST');
req.setEndpoint(ep);

Map<String,String> headerMap = new Map<String,String>();
headerMap.put('Content-Type','text/xml;charset=UTF-8');
headerMap.put('Accept-Encoding','gzip,deflate');
headerMap.put('Content-Length','379');
headerMap.put('HOST','pprojectstest.xxx.ae');
headerMap.put('SOAPAction','http://schemas.microsoft.com/sharepoint/soap/GetItem');
headerMap.put('Connection','Keep-Alive'); 

HttpResponse res = httpObj.send(req,headerMap);

System.debug('Response**:' + res);
System.debug('Response**:' + res.getBody());

And getting the error : System.HttpResponse[Status=Unauthorized, StatusCode=401]

I am not even able to get the challenge in the first reponse with the type1 message. I am able to hit the service though with SOAP UI.

Any help?

**Comment by natewallace** [comment]
Are you running the SOAP UI test from outside of the network that your service is on?

Keep in mind that SalesForce will be calling your service from outside of your network.  Things like firewalls or dns lookups may prevent the service within your network from being reached by applications outside your network if the network isn't configured for this.

**Comment by samirkhan70** [comment]
Yes Nate. I am running the SOAP UI test from outside the netwok and I am able to get the response. 

When making a callout from salesforce , I am not getting the type2 message an getting the error.

If there would have been firewall issue, i  should not get the response in SOAP UI as well.

Any thoughts on this?

**Comment by natewallace** [comment]
Maybe it's related to this issue:  https://github.com/natewallace/ApexNTLM/issues/4
You could try removing the line that clears out the message body when preAuthenticate = true
